ios/BeaconScanner.swift

```diff
@@ -0,0 +1,204 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/* ==!== Heavily edited for the URL discovery use case ==!== */
+
+import UIKit
+import CoreBluetooth
+
+///
+/// BeaconScannerDelegate
+///
+/// Implement this to receive notifications about beacons.
+protocol BeaconScannerDelegate {
+  func urlContextChanged(beaconScanner: BeaconScanner)
+}
+
+///
+/// BeaconScanner
+///
+/// Scans for Eddystone compliant beacons using Core Bluetooth. To receive notifications of any
+/// sighted beacons, be sure to implement BeaconScannerDelegate and set that on the scanner.
+///
+class BeaconScanner: NSObject, CBCentralManagerDelegate {
+  
+  var delegate: BeaconScannerDelegate?
+  
+  private var centralManager: CBCentralManager!
+  private let beaconOperationsQueue: dispatch_queue_t = dispatch_queue_create("beacon_operations_queue", nil)
+  private var shouldBeScanning: Bool = false
+  private var _urlFound = Set<NSURL>()
+  
+  var urlFound: Set<NSURL> {
+    get {
+      return _urlFound
+    }
+  }
+  
+  var urls: Array<String> {
+    get {
+      var urls = Set<String>();
+      for url in _urlFound {
+        urls.insert("\(url)");
+      }
+      return Array(urls);
+    }
+  }
+  
+  override init() {
+    super.init()
+    
+    self.centralManager = CBCentralManager(delegate: self, queue: self.beaconOperationsQueue)
+    self.centralManager.delegate = self
+    
+  }
+  
+  func startScanning() {
+    dispatch_async(self.beaconOperationsQueue) {
```

Comment:
Where does that first arg (`self.beaconOperationsQueue`) come from? Just interested :)